src/wasm/wasm-type.cpp

```diff
@@ -3306,13 +3659,39 @@ std::optional<TypeBuilder::Error> canonicalizeIsorecursive(
     return {*error};
   }
 
-  // Move the recursion groups into the global store. TODO: after proper
-  // isorecursive canonicalization, some groups may no longer be used, so they
-  // will need to be filtered out.
-  std::lock_guard<std::mutex> lock(recGroupsMutex);
-  for (auto& info : recGroupInfos) {
-    recGroups.emplace_back(std::move(info));
+  // Now that we know everything is valid, start canonicalizing recursion
+  // groups. Before canonicalizing each group, update all the use sites with it
+  // to make sure it only refers to other canonical groups or to itself. To
```

Comment:
```suggestion
  // to make sure it only refers to other canonical groups or to itself. That
  // is, it may have references to groups that have been canonicalized
  // with something before them; we want to only refer to the canonical
  // group for all things equal to it. To
```

Reply:
I added some different text to this comment before I saw your suggestion. Let me know if it makes sense!

Reply:
Your text lgtm!